backend/users/managers.py

```diff
@@ -0,0 +1,27 @@
+from django.contrib.auth.base_user import BaseUserManager
+
+
+class CustomUserManager(BaseUserManager):
+    use_in_migrations = True
+
+    def create_user(
+        self, username, password, first_name, last_name, email, **extra_fields
+    ):
+        if not username:
+            raise ValueError("The given username must be set")
+        user = self.model(
+            username=username,
+            first_name=first_name,
+            last_name=last_name,
+            email=email,
+            **extra_fields
+        )
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, username, password, **extra_fields):
+        extra_fields.setdefault("is_staff", True)
+        extra_fields.setdefault("is_superuser", True)
+
+        return self.create_user(username, password, **extra_fields)
```

Comment:
I'm not sure if it's necessary to inherit from `AbstractBaseUser`, since I don't see complex user creation behaviors here.
Maybe it would be better to inherit from `AbstractUser` instead, as it provides `User`'s behavior out of the box. What is more, creators themselves advice to create separate model for storing additional user info (https://docs.djangoproject.com/en/3.2/topics/auth/customizing/#extending-django-s-default-user)